open_lm/main.py

```diff
@@ -729,12 +750,13 @@ def main(args):
             completed_epoch,
             evaluation_loss,
             step=global_step,
+            is_final_checkpoint=final_epoch or (completed_epoch == args.epochs),
```

Comment:
Can be a follow up PR, but should we just set args.epoch to completed_epoch earlier (and emit a warning) instead of creating an auxiliary final_epoch variable?

Reply:
setting args.epoch a bit dicey as we are using this var in an outer for loop. but agree this extra var prob is not the best either. we can do something different in a later PR